discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -0,0 +1,324 @@
+        except Exception as e:
+            logger.error(
+                f"index_user_bank.py | Error fetching tx {tx_sig}, {e}", exc_info=True
+            )
+        retries -= 1
+        logger.error(f"index_user_bank.py | Retrying tx fetch: {tx_sig}")
+    raise Exception(f"Failed fetching {tx_sig}, exit loop")
+
+
+def refresh_user_balance(session, redis, user_bank_acct):
+    user_id: Optional[Tuple[int, None]] = (
+        session.query(User.user_id)
+        .join(
+            UserBankAccount,
+            and_(
+                UserBankAccount.bank_account == user_bank_acct,
+                UserBankAccount.ethereum_address == User.wallet,
+            ),
+        )
+        .filter(User.is_current == True)
+        .one_or_none()
+    )
+    logger.info(f"index_user_bank.py | Refresh user_id = {user_id}, {user_bank_acct}")
+    # Only refresh if this is a known account within audius
+    if user_id:
+        logger.info(f"submit user to queue: {user_id[0]}")
+        enqueue_balance_refresh(redis, [user_id[0]])
+
+
+def process_user_bank_tx_details(session, redis, tx_info, tx_sig, timestamp):
+    meta = tx_info["result"]["meta"]
+    error = meta["err"]
+    if error:
+        logger.info(
+            f"index_user_bank.py | Skipping error transaction from chain {tx_info}"
+        )
+        return
+    account_keys = tx_info["result"]["transaction"]["message"]["accountKeys"]
+    instructions = meta["logMessages"]
+    for msg in instructions:
+        if "EthereumAddress" in msg:
+            public_key_str, public_key_bytes = parse_eth_address_from_msg(msg)
+            logger.info(f"index_user_bank.py | {public_key_str}")
+            # Rederive address based on user public key
+            _, derived_address = get_address_pair(
+                WAUDIO_PROGRAM_PUBKEY, public_key_bytes, USER_BANK_KEY, SPL_TOKEN_ID
+            )
+            bank_acct = str(derived_address[0])
+            # Confirm expected address is present in transaction
+            try:
+                bank_acct_index = account_keys.index(bank_acct)
+                session.add(
+                    UserBankAccount(
+                        signature=tx_sig,
+                        ethereum_address=public_key_str,
+                        bank_account=bank_acct,
+                        created_at=timestamp,
+                    )
+                )
+                if bank_acct_index:
+                    logger.info(
+                        f"index_user_bank.py | Found known account: {public_key_str}, {bank_acct}"
+                    )
+            except ValueError as e:
+                logger.info(e)
+        elif "Transfer" in msg:
+            # Accounts to refresh balance
+            acct_1 = account_keys[1]
+            acct_2 = account_keys[2]
+            logger.info(
+                f"index_user_bank.py | Balance refresh accounts: {acct_1}, {acct_2}"
+            )
+            refresh_user_balance(session, redis, acct_1)
```

Comment:
Just sanity checking here - what if we encounter a user bank that we don't have a corresponding eth address for? Is this going to crash us anywhere (i.e. we assume we have it)?

Reply:
That shouldn't ever happen because we should see the event for creating the user_bank_account first before the transfer events.
If it does happen, this is a no-op and no user is enqueued to be refreshed.